File: typescript/src/payment/processors.ts
/**
 * Payment processors - External service integrations
 */

export function creditCardProcessor(cardNumber: string, cvv: string, expiry: string, amount: number): boolean {
  // Simulate network delay
  // Note: In real code this would be async
  
  // Simulate random failures
  if (Math.random() < 0.1) { // 10% failure rate
    throw new Error("Credit card processing failed");
  }
  
  return true;
}

export function paypalProcessor(email: string, amount: number): boolean {
  // Simulate network delay
  // Note: In real code this would be async
  
  // Simulate random failures
  if (Math.random() < 0.1) { // 10% failure rate
    throw new Error("PayPal processing failed");
  }
  
  return true;
}

File: python/src/payment/processors.py
"""
Payment processors - External service integrations
"""

import random
import time

def credit_card_processor(card_number, cvv, expiry, amount):
    """Simulate credit card processing"""
    # Simulate network delay
    time.sleep(0.1)
    
    # Simulate random failures
    if random.random() < 0.1:  # 10% failure rate
        raise Exception("Credit card processing failed")
    
    return True


def paypal_processor(email, amount):
    """Simulate PayPal processing"""
    # Simulate network delay
    time.sleep(0.1)
    
    # Simulate random failures
    if random.random() < 0.1:  # 10% failure rate
        raise Exception("PayPal processing failed")
    
    return True

